src/erl_optics.erl

```diff
@@ -105,22 +82,13 @@ gauge_set(Key, Val) ->
     case get_lens(Key) of
         {ok, Ptr} ->
             erl_optics_nif:gauge_set(Ptr, Val);
+        {error, {key_not_found, _}} ->
+            {ok, Ptr} = gauge_set_alloc(Key, Val),
```

Comment:
Is there a race condition possible here? I suppose you'll spam the server until foil is reloaded?

Reply:
Solved by ets storing and checking before sending stuff to the server.